lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -31,24 +31,30 @@ def query_api(
             except DatasetNotFound:
                 dataset_id = dataset  # Allow passing deliberately bad value
             headers = {"authorization": f"bearer {pbench_drb_token}"}
-            k = "" if target is None else f"/{target}"
             response = client.get(
-                f"{server_config.rest_uri}/datasets/{dataset_id}/inventory{k}",
+                f"{server_config.rest_uri}/datasets/{dataset_id}/inventory/{target}",
                 headers=headers,
             )
             assert response.status_code == expected_status
             return response
 
         return query_api
 
-    def mock_find_dataset(self, dataset):
-        class Tarball(object):
-            unpacked = Path("/dataset/")
-            tarball_path = Path("/dataset/tarball.tar.xz")
+    class Tarball:
+        def __init__(self, path: Path, controller: Controller):
```

Comment:
This is intended to be a mock which replaces the real `Tarball` class, right?  I think it would be better if it had a distinguished name, to avoid possibly confusion or conflicts with the real class...something like `MockTarball`.